### ISSUE TITLE
Exclude list delete files from both side

### ISSUE THREAD

**Issue by goDamri** [deajan/osync#201]
**Describe the bug**
Tried to add some list to exclude, but i get them deleted from source and destination

**To Reproduce**
RSYNC_EXCLUDE_FROM="exclude.list"

**Additional context**

**Comment by deajan** [comment]
Can you give more details ? Like exactly what you're trying to achieve, osync version used ? config file / command line ?

**Comment by victor-vc** [comment]
Here is an use case that I have experienced during the evaluation stage of the osync tool (master version) that might be related to this issue:
- edit the config file and run the osync.sh (for the initial sync of the target)
- add some exclusions patterns to RSYNC_EXCLUDE_FROM into config file
- run osync.sh again
- the files matching the patterns list will be deleted from both target and initiator.

I think that the matching files should be only deleted (excluded) from the target.

However, if the RSYNC_EXCLUDE_FROM is already configured on the first run (initial sync), the matching files will neither be transferred to nor deleted from master subsequently.

**Comment by deajan** [comment]
Okay I see... That's actually a known issue (see KNOWN ISSUES).
No easy way around this one.

**Comment by victor-vc** [comment]
From what I noticed, this is not strictly related to "soft deletion". The files are deleted regardless of "soft deletion" enabled or disabled.

**Comment by deajan** [comment]
I'm sorry, I have a hard time getting what you're talking about here.
Care to make a quick schema of the behavior you're describing ?

**Comment by deajan** [comment]
I just tried the following:

Create a conf file based on the template like
```
INITIATOR_SYNC_DIR="/tmp/testA"
TARGET_SYNC_DIR="/tmp/testB"

...

RSYNC_EXCLUDE_FROM="exclude.list"
```

Created the following exclude.list file
```
*.test
```

Creating files in directories
```
touch /tmp/testA/{file1.ext,shouldnotbedeleted1.test}
touch /tmp/testB/{file2.ext,shouldnotbedeleted2.test}
```

Run osync

Result: 
```
ls testA
file1.ext  file2.ext  shouldnot1.test

ls testB
file1.ext  file2.ext  shouldnot2.test
```

This is the excpected behavior. Files with extension `.test` haven't been touched nor deleted by osync.
@goDamri, @victor-vc Can you explain the problem you're experiencing this way so I get to know what happens ?

**Comment by victor-vc** [comment]
Here is the scenario (unexpected behavior):

- create the _test.conf_ file
```
INITIATOR_SYNC_DIR="/initiator/folder"
TARGET_SYNC_DIR="ssh://target.host.com//target/folder"
...
SOFT_DELETE=false
```
- add some files to the initiator folder
`touch a.txt b.txt c.test`

- run osync (initial sync)
`osync.sh /etc/osync/test.conf`

**Result**: all 3 files are synced (i.e. existing in both initiator and target folders)

- edit _test.conf_ and add the the exclusion pattern
`RSYNC_EXCLUDE_PATTERN="*.test"`

- run osync again

**Result**: _c.test_ is deleted from both initiator and target.

**Comment by deajan** [comment]
Ok, now it's getting clear.
osync is statefull, which means it keeps directory changes between runs.
If you happen to modify the configuration, you have to reinitialize the statefullness, ie run osync once with `--initialize` parameter.

I'll add a word to the doc.

**Comment by victor-vc** [comment]
Yes, it makes sense because if I add `RSYNC_EXCLUDE_PATTERN="*.test"` in the conf file from the very beginning (prior running the initial sync), the files are preserved on initiator and are not transferred (are excluded) from target.

**Comment by deajan** [comment]
This was freaking scary... A big blocker bug just before a release.
Fortunately, it's just doc updates ;)
